rfc/rfc-63/rfc-63.md

```diff
@@ -0,0 +1,370 @@
+  existing flow of bulk insert and clustering
+    - Each data file contains one or a small subset of values corresponding to the Index Functions.
+    - Different sort modes can be provided like bulk insert and clustering.
+
+- For updates, no change in logic is needed. The record with the same key should go into the same file group.
+- For small file handing, we need to take into account the Index functions for the workload profile, so that the new
+  inserts can be combined into the data file without expanding the minimum and maximum values of the Index Function when
+  the data file is small.
+
+## Related Concepts
+
+In this section, we compare Hudi's Index Function to related concepts and solutions on pruning files efficiently and
+improving query performance.
+
+### Hidden Partitioning
+
+[Hidden Partitioning](https://iceberg.apache.org/docs/latest/partitioning/#icebergs-hidden-partitioning), introduced by
+Apache Iceberg, maintains the relationship between the original column and transformed partition values, and hides the
+partition columns and physical partitions from catalogs and users, using manifests containing the partition values of
+each file to prune files and speed up queries. By using Hidden Partitioning, Iceberg can evolve partition schemes, i.e.,
+partition evolution, without rewriting data.
+
+Hudi's Index Function achieves the same goal of pruning files based on the column stats, with the support on any data
+column or function defined on a data column, without coupling with the partition scheme.
+
+### Index on Expression
+
+[Index on Expression](https://www.postgresql.org/docs/current/indexes-expressional.html),
+or [Function-Based Indexes](https://oracle-base.com/articles/8i/function-based-indexes) a concept in RDBMS, provides the
+index on a function or scalar expression computed from one or more columns of the table. Such an index can speed up the
+queries containing the expression in the predicates. The index on expression is a secondary index, meaning that it is
+stored separately from the record data. There is no materialization of the values from the expression in the data area.
+PostgreSQL supports Index on Expression and Oracle database supports Function-Based Indexes.
+
+Hudi's Index Function applies similar concept which allows a new index to be built based on a function defined on a data
+column at any time for file pruning.
+
+### Generated Column
+
+A [Generated Column](https://www.postgresql.org/docs/15/ddl-generated-columns.html) is a special column that is always
+computed from other columns. There are two types of generated column: (1) stored generated column: written alongside
+other columns and persisted to storage, (2) virtual generated column: computed when read and occupies no storage.
+PostgreSQL and Delta Lake support stored generated column only. As data of stored generated column is persisted to
```

Comment:
I think we should make it clear that these generated columns are materialized in the data files themselves (as compared to Functional Index above)